### PR TITLE
docs(python): simplify/clarify group_by_dynamic examples

### PR DESCRIPTION
**Pull request by MarcoGorelli**
To be honest I think the current examples are bit too complicated - they can be simplified to (hopefully better) get the point across

The rows are already kind of numbered by `'n'`, by just aggregating by `pl.col('n')` it becomes clear which row belongs to which window

And it's now easier to read without datetimes all of the place

### CODE REVIEW

File: py-polars/polars/lazyframe/frame.py [hunk 1]
```diff
@@ -3235,13 +3216,15 @@ def group_by_dynamic(
...
-        ... ).agg(pl.col("A").alias("A_agg_list")).collect()
+        >>> (
+        ...     lf.group_by_dynamic(
+        ...         "idx",
+        ...         every="2i",
+        ...         period="3i",
+        ...         include_boundaries=True,
+        ...         closed="right",
+        ...     ).agg(pl.col("A").alias("A_agg_list"))
+        ... ).collect()
```
Comment (stinodego) [inline py-polars/polars/lazyframe/frame.py:3227]: Looks like something messed up the formatting here?

**Review by stinodego — COMMENTED** [review]
Much better! Just one minor comment on formatting.

File: py-polars/polars/lazyframe/frame.py (hunk 1 above)
Comment (MarcoGorelli) [reply py-polars/polars/lazyframe/frame.py]: yup, thanks 👍

File: py-polars/polars/lazyframe/frame.py [hunk 2]
```diff
@@ -3241,7 +3222,7 @@ def group_by_dynamic(
         ...     period="3i",
         ...     include_boundaries=True,
         ...     closed="right",
-        ... ).agg(pl.col("A").alias("A_agg_list")).collect()
+        ... ).agg(pl.col("A").alias("A_agg_list"))
```
Comment (MarcoGorelli) [inline py-polars/polars/lazyframe/frame.py:3225]:
```suggestion
        ... ).agg(pl.col("A").alias("A_agg_list")).collect()
```

**Review by stinodego — APPROVED** [review]

**Pull request merged**